### DISCUSSION TITLE
Web reader + progress sync + joplin

### DISCUSSION THREAD

**Discussion by rolocampusano** [koreader/koreader#8249] [Ideas]
Just had a thought. What about creating a docker image file that includes guacamole and the linux reader. Then, i should be able read in a website, right? :)

If I do this + sync my books with dropbox/anything and use the koreder progress sync plugin. Then I should be able to read sync my readings between my e-reader and the web, right? 

I should also be able to use joplin locally (in the docker image if needed) and then sync my notes and highlights automatically. 

What do you guys think about an implementation like this? Or is it an overkill?

**Comment by Frenzie** [comment]
I would say that sounds inferior compared to using some other native program but it also sounds like something you could probably set up in a few minutes if you feel differently? ;-) (Assuming there's a Guacamole image where all you need to do is to add the AppImage on top of it.)

**Reply by rolocampusano** [reply]
Thanks for the reply! What would be a native program that would make something similar? I really want to be able to sync highlights, notes and progress across devices.

**Reply by Frenzie** [reply]
I'm not sure. It just doesn't sound like the best experience to me. ^_^

**Reply by rolocampusano** [reply]
Uff. I completely agree. I wonder if there is a way to create a web app that can be self-hosted instead. I currently use the calibre-web reader but it doesnt sync :(.

**Reply by ryanwwest** [reply]
The web app seems neat. Won't likely conflict with browser standard keyboard shortcuts since there aren't many to begin with. Then you could potentially access a 'KOReader' local webpage from any one of your devices to read your entire library (that's hosted on the server device). Likely inferior to the native app, but the convenience opens up a lot of options for reading.

Of course some plugins/features, like the terminal, may not work this way. You'd probably need to set up authentication and maybe users for something concrete.
